File: Java-Macyna/tydzien_2/Zadanie_1/LiczbyPierwsze.java
public class LiczbyPierwsze {
    private int pierwsze[];

    public LiczbyPierwsze (int n) {
        pierwsze = new int[n];
        int miejsce = 0;

        for(int i = 2; i <= n; i += 1) {
            int licznik = 0;
            for(int j = 1; j * j <= i; j += 1) {
                if(i % j == 0){
                    licznik += 1;
                    if(licznik == 2) {
                        break;
                    }
                }
            }
            if(licznik < 2) {
                    pierwsze[miejsce] = i;
                    miejsce += 1;
            }
        }
    }

    public int liczba(int m) {
        return pierwsze[m];
    }
}

File: Java-Macyna/tydzien_2/Zadanie_2/LiczbyPierwsze.cpp
//LiczbyPierwsze.cpp

#include <iostream>
#include "nagl.hpp"

LiczbyPierwsze::LiczbyPierwsze(int n) {
    pierwsze = (int*) malloc(n * sizeof(int));
    int indeks = 0;

    for(int i = 2; i <= n; i += 1) {
        bool isPrime = true;
        for(int j = 2; j * j <= i; j += 1) {
            if(i % j == 0) {
                isPrime = false;
                break;
            }
        }
        if(isPrime == true) {
            *(pierwsze + (indeks * 4)) = i;
            indeks += 1;
        }
    }
}

LiczbyPierwsze::~LiczbyPierwsze() {
    free(pierwsze);
}

int LiczbyPierwsze::liczba(int n) {
    return *(pierwsze + (n * 4));
}